### PR TITLE
Add APIScan to build

### LINKED ISSUE #7085: Enable APIScan on all our built binaries
In order to ensure compliance with the APIs our libraries call we need to enable the APIScan tool for any binaries we release. The tool only supports .NET and Native built binaries so this will need to be enabled for our .NET builds and the TypeSpec and azd tool builds. For languages like TS, Python, and Java which don't build down to native binaries we don't have any tools to verify those. For languages like Go, C, and C++ while we don't ship binaries we do build them in our test builds so we should run the tool against the binaries in one of our test builds.

For more information about APIScan see https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/32485/APIScan and https://eng.ms/docs/products/apiscan

### PR DESCRIPTION
**Pull request by weshaggard**
Info on task https://aka.ms/gdn-azdo-apiscan

Fixes https://github.com/Azure/azure-sdk-tools/issues/7085

### CODE REVIEW

**Comment by weshaggard** [comment]
OK this is now ready to merge. The other failures in aggregate-reports are unrelated to ApiScan.

File: eng/pipelines/aggregate-reports.yml [hunk 1]
```diff
@@ -156,4 +156,46 @@ stages:
...
+              softwareFolder: $(Build.ArtifactStagingDirectory)
+              # SoftwareName cannot change as it is used for looking up baselining data
+              softwareName: 'Azure SDK Libraries'
+              softwareVersionNum: '$(OfficialBuildId)'
+              isLargeApp: false
+              preserveLogsFolder: true
+              verbosityLevel: standard
+            env:
+              # azure-sdk-apiscan (81109e5f-0620-423c-a37a-c22fbf8973a7)
+              AzureServicesAuthConnectionString: runAs=App;AppId=81109e5f-0620-423c-a37a-c22fbf8973a7;TenantId=72f988bf-86f1-41af-91ab-2d7cd011db47;AppKey=$(azure-sdk-apiscan-client-secret)
```
Comment (benbp) [inline eng/pipelines/aggregate-reports.yml:187]: Would it be easier to wrap the whole conn string into the secret? That way if we change the app we don't have to update it in two places.

**Review by benbp — APPROVED** [review]

File: eng/pipelines/aggregate-reports.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/aggregate-reports.yml:187]: I thought about that, and I was trying to keep just the secret as opposed to a full configuration. I think of this as no different than a username which we generally don't put into KV.

[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/pipelines/aggregate-reports.yml:187]: These are just identifiers similar to a username and aren't secrets, so they don't need to be protected the same way as the key.

**Pull request merged**